arbor/include/arbor/recipe.hpp

```diff
@@ -69,7 +69,7 @@ struct gap_junction_connection {
 
     gap_junction_connection(cell_global_label_type peer, cell_local_label_type local, double g):
         peer(std::move(peer)), local(std::move(local)), weight(g) {
-        if (std::isnan(weight)) throw std::out_of_range("Gap junction weight must be finite.");
+        if (std::isnan(weight)) throw std::domain_error("Gap junction weight must be finite.");
```

Comment:
Similar comments as above: testing finite or testing Nan? We should use an arbor exception.